Documentation/project-docs/platform-names.md

```diff
@@ -4,6 +4,24 @@ Over the years, we've accumulated quite a few names, platforms, and components t
 
 This will also list some aliases. As you'll see the aliases aren't always correct -- they are merely listed to help you find the better and less confusing terminology.
 
+In this document, the following terms are used:
+
+* **IL**. Intermediate language. IL is a stack-based instruction set and is what higher level languages, such as C#, compile down to.
```

Comment:
:+1:

> IL in .NET is also sometimes called MSIL (Microsoft Intermediate Language), to separate it from other systems with intermediate languages.

RE: "stack-based" ... tough one ... it might take a little too long to explain anything meaningful, but ...

> Higher level .NET languages, such as C#, compile down a stack-based instruction set (simpler for a machine to execute), which is called Intermediate Language (IL).

... or some other parenthetical note that just tells the reader that stack-based must be better for computers to actually use over the instruction set that mere mortals provide.